internal/core/api.rs

```diff
@@ -198,37 +340,37 @@ impl Window {
 
     /// This function returns an euclid scale that allows conveniently converting between logical and
     /// physical pixels based on the window's scale factor.
-    pub fn scale_factor(&self) -> euclid::Scale<f32, LogicalPx, PhysicalPx> {
-        self.0.scale()
+    pub fn scale_factor(&self) -> f32 {
```

Comment:
I'm wondering if this should not return a newtype that can be used to multiply or divide the other types

Reply:
I thought about this as well but I must admit that I've come to really like this API, for two reasons:

1. It turns out that there are still many call sites that want just the scalar value (as you can see in this diff).
2. The conversion between logical and physical coordinates becomes more expressive:

```rust
let size_to_save_and_restore = window.size() / window.scale_factor();

// vs.

let size_to_save_and_restore = window.size().to_logical(window.scale_factor);
```

The `to_logical` is by far more readable that a single character `/`, in a context where it helps to minimise the effort to clarify whether the size variable is in logical or physical coordinate space.